core/Loc.h

```diff
@@ -23,24 +23,32 @@ struct LocOffsets {
     bool exists() const {
         return endLoc != INVALID_POS_LOC && beginLoc != INVALID_POS_LOC;
     }
-} __attribute__((packed, aligned(1)));
-CheckSize(LocOffsets, 6, 1);
+    static LocOffsets none() {
+        return LocOffsets{INVALID_POS_LOC, INVALID_POS_LOC};
+    }
+    LocOffsets join(LocOffsets other) const;
+    // For a given Loc, returns a zero-length version that starts at the same location.
+    LocOffsets copyWithZeroLength() const {
+        return LocOffsets{beginPos(), beginPos()};
+    }
```

Comment:
Does this function still make sense here? Maybe it should move to `Loc`, or be inlined into the places that were using it?

Reply:
Happy either way. Do you want it done as part of this PR?

Reply:
No it's fine to do it later or not do it.